pkg/aws/eks/eks.go

```diff
@@ -0,0 +1,524 @@
+	return "eks"
+}
+
+func NewCollector(region string, profile string, scrapeInternal time.Duration, ps *pricing.Client, ec2s *ec2.Client, profiles []string) (*Collector, error) {
+	return &Collector{
+		Region:         region,
+		Profile:        profile,
+		Profiles:       profiles,
+		ScrapeInterval: scrapeInternal,
+		pricingService: ps,
+		ec2Client:      ec2s,
+	}, nil
+}
+
+func (c *Collector) Register(_ provider.Registry) error {
+	return nil
+}
+
+func (c *Collector) ListOnDemandPrices(ctx context.Context, region string, client *pricing.Client) ([]string, error) {
+	var productOutputs []string
+	input := &pricing.GetProductsInput{
+		ServiceCode: aws.String("AmazonEC2"),
+		Filters: []types.Filter{
+			{
+				Field: aws.String("regionCode"),
+				// TODO: Use the defined enum for this once I figure out how I can import it
+				Type:  "TERM_MATCH",
+				Value: aws.String(region),
+			},
+			{
+				// Limit output to only base installs
+				Field: aws.String("preInstalledSw"),
+				Type:  "TERM_MATCH",
+				Value: aws.String("NA"),
+			},
+			{
+				// Limit to shared tenancy machines
+				Field: aws.String("tenancy"),
+				Type:  "TERM_MATCH",
+				Value: aws.String("shared"),
+			},
+			{
+				// Limit to compute instances(ie, not bare metal)
+				Field: aws.String("productFamily"),
+				Type:  "TERM_MATCH",
+				Value: aws.String("Compute Instance"),
+			},
+			{
+				// RunInstances is the operation that we're interested in.
+				Field: aws.String("operation"),
+				Type:  "TERM_MATCH",
+				Value: aws.String("RunInstances"),
+			},
+			{
+				// This effectively filters only for ondemand pricing
+				Field: aws.String("capacitystatus"),
+				Type:  "TERM_MATCH",
+				Value: aws.String("UnusedCapacityReservation"),
+			},
+			{
+				// Only care about Linux. If there's a request for windows, remove this flag and expand the pricing map to include a key for operating system
+				Field: aws.String("operatingSystem"),
+				Type:  "TERM_MATCH",
+				Value: aws.String("Linux"),
+			},
+		},
+	}
+
+	for {
+		products, err := client.GetProducts(ctx, input)
+		if err != nil {
+			return productOutputs, err
+		}
+
+		if products == nil {
+			break
+		}
+
+		productOutputs = append(productOutputs, products.PriceList...)
+		if products.NextToken == nil {
+			break
+		}
+		input.NextToken = products.NextToken
+	}
+	return productOutputs, nil
+}
+
+func ListComputeInstances(ctx context.Context, region string, profile string) ([]ec2Types.Reservation, error) {
+	options := []func(*awsconfig.LoadOptions) error{awsconfig.WithEC2IMDSRegion()}
+	options = append(options, awsconfig.WithRegion(region))
+	options = append(options, awsconfig.WithSharedConfigProfile(profile))
+	ac, err := awsconfig.LoadDefaultConfig(context.Background(), options...)
+	if err != nil {
+		return nil, err
+	}
+	client := ec2.NewFromConfig(ac)
+	dii := &ec2.DescribeInstancesInput{
+		// TODO: Is 1000 appropriate?
+		MaxResults: aws.Int32(1000),
+	}
```

Comment:
I wonder if something like this may improve performance?

```suggestion
	dii := &ec2.DescribeInstancesInput{
		// TODO: Is 1000 appropriate?
		MaxResults: aws.Int32(1000),
		Filters:    []ec2Types.Filter{{Name: aws.String("tag-key"), Values: clusterTags}, {}},
	}
```

This way, only ec2 instances that have the one of the cluster tags [(values are OR'd)](https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/ec2@v1.161.1/types#Filter) will be returned?

Reply:
Right now we're able to process the compute instances in prod in ~10s. Let me add it and see what happens 🤔

Reply:
I'm going to run this a bit longer tomorrow and report back. IE, let me run it for ~30 minutes with the filter and 30 minutes without so I can get a proper comparison.

Reply:
![image](https://github.com/grafana/cloudcost-exporter/assets/1080566/5bee74e6-ab15-418c-bb70-fb24bc9b90d0)
The first half of the time series is with the filter suggested, and the last half is without the filter. It's really hard to tell, but it doesn't make an obvious improvement in speed. I'm going to leave out the filter for now, but leave a comment for the future that it's possible to filter by tags and to benchmark it.